### PR TITLE
Implement podman-remote umount and rm command

### PR DESCRIPTION
**Pull request by jwhonce**
* refactor command output to use one function
* Add new worker pool parallel operations
* Implement podman-remote umount
* Refactored podman wait to use printCmdOutput()

Carries commits from https://github.com/containers/libpod/pull/2549

Signed-off-by: Jhon Honce <jhonce@redhat.com>

### CODE REVIEW

**Comment by rhatdan** [comment]
LGTM

**Comment by rhatdan** [comment]
LGTM

File: cmd/podman/main.go [hunk 1]
```diff
@@ -48,6 +48,7 @@ var mainCommands = []*cobra.Command{
 	podCommand.Command,
 	_pullCommand,
 	_pushCommand,
+	_rmCommand,
```
Comment (mheon) [inline cmd/podman/main.go]: Should unmount be moved as well?

File: cmd/podman/shared/workers.go [hunk 2]
```diff
@@ -0,0 +1,126 @@
+package shared
```
Comment (mheon) [inline cmd/podman/shared/workers.go]: Did this move from somewhere? Can we move all instances to use this and delete the old code?

File: cmd/podman/rm.go [hunk 3]
```diff
@@ -52,148 +48,30 @@ func init() {
...
 
-	ctx := getContext()
-	runtime, err := libpodruntime.GetRuntime(&c.PodmanCommand)
+	runtime, err := adapter.GetRuntime(&c.PodmanCommand)
 	if err != nil {
 		return errors.Wrapf(err, "could not get runtime")
 	}
 	defer runtime.Shutdown(false)
 
-	if rootless.IsRootless() {
```
Comment (mheon) [inline cmd/podman/rm.go]: Did we lose all the rootless handling here? That's going to break a lot of things.

Comment (baude) [reply cmd/podman/rm.go]: it looks like he's just moved it to be dealt with where he gets the containers right @jwhonce

**Comment by baude** [comment]
address @mheon comments and test failures and LGTM! well done

File: cmd/podman/rm.go (hunk 3 above)
Comment (mheon) [reply cmd/podman/rm.go]: But that doesn't work? The rootless logic is different for different commands - I think @giuseppe just overhauled the way `rm` joins namespaces to fix some issues.

@giuseppe can you confirm that?

File: cmd/podman/main.go (hunk 1 above)
Comment (jwhonce) [reply cmd/podman/main.go]: Good catch, done.

File: cmd/podman/rm.go (hunk 3 above)
Comment (jwhonce) [reply cmd/podman/rm.go]: The presentation level can no longer do this kind of work as it's run on both the client and server.

I moved it into shortcuts.GetContainerByContext().  I'll be happy to work with @giuseppe if that is wrong.  But his last commit won't work with podman-remote.

**Comment by TomSweeneyRedHat** [comment]
Changes LGTM , but you need to rebase.

**Comment by rhatdan** [comment]
@jwhonce Needs a rebase.

**Comment by baude** [comment]
@jwhonce way to stick with it! good show!

**Pull request merged**